data-prepper-core/src/main/java/org/opensearch/dataprepper/sourcecoordination/LeaseBasedSourceCoordinator.java

```diff
@@ -189,7 +189,10 @@ public <S extends T> void saveProgressStateForPartition(final String partitionKe
 
     @Override
     public void giveUpPartitions() {
-        validateIsInitialized();
+
+        if (!initialized) {
+            return;
+        }
```

Comment:
You should probably add a simple test to fix this code coverage issue.

Reply:
Will add a test for this